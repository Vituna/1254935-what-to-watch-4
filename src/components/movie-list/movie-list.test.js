import React from "react";
import renderer from 'react-test-renderer';

import MovieList from './movie-list.jsx';

const Settings = {
  MOVIE_CARDS: [
    {
      name: `Avatar`,
      image: `img/avatar.jpg`
    },
    {
      name: `Aviator`,
      image: `img/aviator.jpg`
    },
    {
      name: `Bohemian rhapsody`,
      image: `img/bohemian-rhapsody.jpg`
    },
    {
      name: `Johnny English`,
      image: `img/johnny-english.jpg`
    },
    {
      name: `Orlando`,
      image: `img/orlando.jpg`
    },
    {
      name: `Pulp Fiction`,
      image: `img/pulp-fiction.jpg`
    },
    {
      name: `Player poster`,
      image: `img/player-poster.jpg`
    },
    {
      name: `War of the worlds`,
      image: `img/war-of-the-worlds.jpg`
    },
  ]
};

it(`Should WTW render correctly`, () => {
  const tree = renderer
    .create(<MovieList
      movieÐ¡ardsSettings={Settings.MOVIE_CARDS}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

