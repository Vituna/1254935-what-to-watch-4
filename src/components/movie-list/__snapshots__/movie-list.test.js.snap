// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should WTW render correctly 1`] = `
Array [
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
      onClick={[Function]}
    >
      <video
        className="player__video"
        muted={true}
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
        onClick={[Function]}
      />
    </h3>
  </article>,
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
      onClick={[Function]}
    >
      <video
        className="player__video"
        muted={true}
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
        onClick={[Function]}
      />
    </h3>
  </article>,
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
      onClick={[Function]}
    >
      <video
        className="player__video"
        muted={true}
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
        onClick={[Function]}
      />
    </h3>
  </article>,
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
      onClick={[Function]}
    >
      <video
        className="player__video"
        muted={true}
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
        onClick={[Function]}
      />
    </h3>
  </article>,
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
      onClick={[Function]}
    >
      <video
        className="player__video"
        muted={true}
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
        onClick={[Function]}
      />
    </h3>
  </article>,
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
      onClick={[Function]}
    >
      <video
        className="player__video"
        muted={true}
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
        onClick={[Function]}
      />
    </h3>
  </article>,
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
      onClick={[Function]}
    >
      <video
        className="player__video"
        muted={true}
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
        onClick={[Function]}
      />
    </h3>
  </article>,
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
      onClick={[Function]}
    >
      <video
        className="player__video"
        muted={true}
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
        onClick={[Function]}
      />
    </h3>
  </article>,
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
      onClick={[Function]}
    >
      <video
        className="player__video"
        muted={true}
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
        onClick={[Function]}
      />
    </h3>
  </article>,
]
`;
